File: water/src/broker/IBrokerMessageHeaders.ts
export const BROKER_HEADERS = {
    CLIENT_ID: "client-id",
    REQUEST_ID: "request-id",
    SOURCE_CLUSTER: "source-cluster",
    TARGET_CLUSTERS: "target-clusters",
};

export interface IBrokerMessageHeaders {

    get clientId(): string;
    get sourceCluster(): string;
    get targetClusters(): Set<string>;
    get requestId(): string;

}

File: latte/src/main/java/gg/beemo/latte/broker/IBrokerMessageHeaders.kt
package gg.beemo.latte.broker

interface IBrokerMessageHeaders {

    val clientId: String
    val sourceCluster: String
    val targetClusters: Set<String> // // Empty = No target restriction
    val requestId: String

    companion object {

        const val HEADER_CLIENT_ID = "client-id"
        const val HEADER_REQUEST_ID = "request-id"
        const val HEADER_SOURCE_CLUSTER = "source-cluster"
        const val HEADER_TARGET_CLUSTERS = "target-clusters"

    }

}
